modules/gem5/_posts/2000-03-01-assignment1.md

```diff
@@ -130,7 +121,44 @@ Use `4GHz` as the clock frequency.
 
 In your report, answer the same questions after simulation supported with data. A complete set of simualtion data for this step should include **6 configurations** (2 options for CPU model * 3 options for memory model).
 
-### Step III: General questions
+### Step III: Using different compiler optimizations
+
+In this step, you'll be using different compiler optimizations to run the same
+matrix multiplication program. Compiler optimization flags are options that can
+be used to improve the *performance* of the program at the expense of
+compilation time and the ability to debug the program. The default version of
+the matrix multiply program is compiled with the flag `-O2`:
+
+```sh
+g++ -o mm mm.cpp -static -O2
+```
+
+We have compiled binaries with flags `-O0` and `-O3`. The former reduces the
+compilation time and makes debugging produce expected results. The latter
+optimizes the code to the maximum possible limit. This *usually* means that
+there are less number of instructions in the more optimized version of the
+program. Let's look at the following computation: $z = \alpha x + y$.
+This operation first multiplies $\alpha$ and $x$, and then adds $y$ to the
+result. Assume that x, y and z are all float. In modern processors,
+these two steps are usually combined to do a `multiply and accumulate`
+operation in a single step. If the compiler detects such computations, it'll
```

Comment:
```suggestion
these two steps are usually combined in a single `multiply and accumulate`
operation. If the compiler detects such computations, it'll
```

(Note: this is why one sentence per line is important. See how this one small change I made causes cascading changes if you use hard wrapping? It makes reviewing quite difficult.)

Reply:
i'll fix it. by the way, are we following the 79-character limit per line for this repo?

Reply:
No, and we *never* follow that for markdown files.

Reply:
okay. i'll update the assignment's readme as well.